packages/grafana-e2e/src/flows/addDashboard.ts

```diff
@@ -156,25 +157,38 @@ const addVariable = (config: PartialAddVariableConfig, isFirst: boolean): AddVar
 
   // This field is key to many reactive changes
   if (type !== VARIABLE_TYPE_QUERY) {
-    e2e.pages.Dashboard.Settings.Variables.Edit.General.generalTypeSelect().select(type);
+    e2e.pages.Dashboard.Settings.Variables.Edit.General.generalTypeSelect()
+      .should('be.visible')
+      .within(() => {
+        e2e.components.Select.singleValue().should('have.text', 'Query').click().type(`${type}{enter}`);
+      });
   }
 
-  // Avoid '', which is an accepted value
-  if (hide !== undefined) {
-    e2e.pages.Dashboard.Settings.Variables.Edit.General.generalHideSelect().select(hide);
+  // Hide only does not show up for VARIABLE_TYPE_CONSTANT
+  // Hide can be '', so we are explicitly checking for undefined
+  if (type !== VARIABLE_TYPE_CONSTANT && hide !== undefined) {
+    e2e.pages.Dashboard.Settings.Variables.Edit.General.generalHideSelect()
+      .should('be.visible')
+      .within(() => {
+        e2e.components.Select.input().should('have.text', '').type(`${hide}{enter}`);
+      });
```

Comment:
Are there any plugins tests that use this? I think we should remove this option to minimize the length of this flow.

Reply:
:+1:

Reply:
Not at the moment, it looks like our plugins always test the `Constant` type route and prior to v7.4.0, the `hide` option would show up for `Constant` but not anymore, so I'll remove it 😄 

If we need it later, we can split this up into smaller flows (like you mentioned in your comment) 👍